lua/neogit/popups/remote/actions.lua

```diff
@@ -9,20 +9,31 @@ local status = require("neogit.status")
 local FuzzyFinderBuffer = require("neogit.buffers.fuzzy_finder")
 local RemoteConfigPopup = require("neogit.popups.remote_config")
 
-function M.add(popup)
+local operation = require("neogit.operations")
+
+M.add = operation("add_remote", function(popup)
   local name = input.get_user_input("Remote name: ")
   if not name then
     return
   end
 
   local origin = git.config.get("remote.origin.url").value
-  local host, remote = origin:match("([^:/]+)[^/]+(.-%.git)")
-  local remote_url = input.get_user_input("Remote url: ", host .. ":" .. name .. remote)
+  local host, _, remote = origin:match("([^:/]+)[:/]([^/]+)/(.+)")
+
+  remote = remote and remote:gsub("%.git$", "")
+
+  local msg
+  if host and remote then
+    msg = string.format("%s:%s/%s.git", host, name, remote)
+  end
+
+  local remote_url = input.get_user_input("Remote url: ", msg)
```

Comment:
This is ok if `msg` is nil?

Reply:
Yes, nil just means that no default value is provided. This occurs if there is no remote *origin*ally.